include/LightGBM/c_api.h

```diff
@@ -726,6 +726,45 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSR(BoosterHandle handle,
                                                 int64_t* out_len,
                                                 double* out_result);
 
+/*!
+ * \brief Make sparse prediction for a new dataset in CSR format.  Currently only used for feature contributions.
+ * \note
+ * The outputs are pre-allocated, as they can vary for each invocation, but the shape should be the same:
```

Comment:
unlike other methods, the user can't pre-allocate because they can't anticipate what the size of the indices and data will be for the sparse result.  In that case, is it reasonable for us to do the sparse matrix allocation within the method?

Reply:
it is okay, but aware to avoid memory leak.

Reply:
BTW, these new apis are for the sparse output? Maybe we can use a clearer name to indicate this.

Reply:
sure, maybe just:
```
LGBM_BoosterPredictSparseOutput
```
since we are going to unify the CSR and CSC APIs